dagfactory/dagfactory.py

```diff
@@ -33,13 +33,33 @@ def __init__(self, config_filepath: Optional[str] = None, config: Optional[dict]
         self.dags_count: int = 0
         self.tasks_count: int = 0
         self.taskgroups_count: int = 0
+        self._config_filepath = config_filepath
         assert bool(config_filepath) ^ bool(config), "Either `config_filepath` or `config` should be provided"
         if config_filepath:
             DagFactory._validate_config_filepath(config_filepath=config_filepath)
             self.config: Dict[str, Any] = DagFactory._load_config(config_filepath=config_filepath)
         if config:
             self.config: Dict[str, Any] = config
 
+    @staticmethod
+    def _serialise_config_md(dag_name, dag_config, default_config):
+        # Remove empty task_groups if it exists
+        if dag_config.get("task_groups") == {}:
+            del dag_config["task_groups"]
+
```

Comment:
Is DAG factory injecting this empty task group on behalf of end-users?

Reply:
Added a comment: https://github.com/astronomer/dag-factory/pull/305/commits/08c52a2c2337d5dc8c23f3b169d8cecc4ddf49c9